tools/parse-ramdisk-logs.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+
+# extraction datetime is always irrelevant
+EXTRACTION_TS = re.compile(r"^[0-9TZ\.:\-]* ?")
+CONTENTS = re.compile(r"\*{12} Contents of (?P<path>/.*)\.tar\.gz ")
+ENTRY = re.compile(r"\*{4} Entry: (?P<entry>.*) \*{4}")
+
+
+def log(msg):
+    print(msg, file=sys.stderr)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("source")
+parser.add_argument("destination")
+args = parser.parse_args()
+
+# Visitor is an object that is currently responsible for handling new lines.
+# These objects are hierarchical: top level is host (Ironic node), second level
+# is specific source (file or command). Initially, visitor is None: such lines
+# are skipped.
+visitor = None
+
+# Skipped is the counter for initially skipped lines.Unfortunately, this may
+# include not just the inotify chatter but also useful lines if the logs have
+# been rotated. There is nothing that can be done here: the file name is only
+# available in the beginning of a dump.
+skipped = 0
+
+
+class VisitorEntry:
+    def __init__(self, dest, entry):
+        assert entry is not None
+        path = os.path.join(dest, entry)
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        self.file = open(path, "wt")
+
+    def close(self):
+        self.file.close()
+
+    def __call__(self, line):
```

Comment:
typehints, and this line is redefining the line variable from L98. Scope issue should be solved by adding the `if __name__ == "__main__"`  stuff for the top-level code.

```suggestion
    def __call__(self, line: str):
```

Reply:
I can fix that, but note that an `if` condition does not create a new scope (unlike in Go and most compiled languages), so your solution won't improve anything. It has to be a new function.

Reply:
That is true, I did not want to say "also create def main()" but that is of course required part of this. I'm bad at half nitting :)